### PR TITLE
Add Location and group id for Follower Replication

### PR DESCRIPTION
**Pull request by Fullstop000**
Part of https://github.com/pingcap/community/issues/86

Add a new type `Location` to represent the geoinformation of a store in `metapb.proto`
Add a new field `group_id` in `RaftMessage` to represent the Group of `from_peer` in Follower Replication 

PTAL @hicqu @BusyJay 

Signed-off-by: Fullstop000 <fullstop1005@gmail.com>

### CODE REVIEW

File: proto/metapb.proto [hunk 1]
```diff
@@ -44,6 +51,8 @@ message Store {
     // Status address provides the HTTP service for external components
     string status_address = 7;
     string git_hash = 8;
+    // Indicates the location the store is at.
+    Location location = 9;
```
Comment (hicqu) [inline proto/metapb.proto]: I think there are 2 solution for this:
1 is configuring `group-id`s with stores, just like here. So leader can know the associated `group-id` for every store by asking PD.
2 is add `group-id` in `RaftMessage`, so leader can know followers' `group-id` after they send any messages to the leader.
PTAL @BusyJay

**Comment by breeswish** [comment]
@nrc Rust build is broken, could you take a look?

**Comment by nrc** [comment]
Whoops, let me send a quick PR...

**Comment by nrc** [comment]
Build is fixed by https://github.com/pingcap/kvproto/pull/505

**Pull request closed without merging**